entity-server-api/src/main/java/org/terracotta/entity/ServiceProvider.java

```diff
@@ -22,46 +22,62 @@
 
 
 /**
- * Service Provider which are provided at the platform level.  A service provider is responsible for instantiating the
- * services to interact with its underlying implementation in response to platform requests.  While the services are
- * generally used by entities, they are also used by the platform, itself.
+ * A service provider is responsible for instantiating the services to interact with its underlying implementation in response
+ *  to platform requests.  While the services are generally used by entities, they are also used by the platform, itself.
+ * 
+ * Note that there are no threading restrictions on how the methods in this interface will be called.  This means that any
+ *  implementation which has shared state must ensure that it can manage this state in a thread safe way (subject to whatever
+ *  thread safety its own state requires).
  */
 public interface ServiceProvider {
-
   /**
-   * The platform configuration based on which the Service provider can choose to initialize itself.
-   *
-   * @param configuration platform configuration
-   * @return true if provider has successfully been initialized with the provided configuration
+   * The platform configuration based on which the receiver can choose to initialize itself.
+   * This is called once on each provider instance, at server start-up time, before any other methods have been called on the
+   *  instance.
+   * Note that it is possible for there to be multiple instances of the same ServiceProvider if the server had multiple
```

Comment:
Based on the ServiceConfiguration that is passed in by the entity when asking for a service.  Each ServiceProvider will be asked if it can satisfy the ServiceConfiguration.  For example, a Service implementor can use a naming scheme and pass a unique name with the ServiceConfiguration.